mypy/util.py

```diff
@@ -384,13 +467,21 @@ def style(self, text: str, color: Literal['red', 'green', 'blue', 'yellow', 'non
             start = ''
         if underline:
             start += self.UNDER
+        if dim:
+            start += self.DIM
         return start + self.colors[color] + text + self.NORMAL
 
     def colorize(self, error: str) -> str:
         """Colorize an output line by highlighting the status and error code."""
         if ': error:' in error:
             loc, msg = error.split('error:', maxsplit=1)
-            if not self.show_error_codes:
+            if self.show_source_code:
+                # Improve readability by wrapping lines when showing source code.
+                pad = len(loc) + len('error: ')
+                max_len = get_term_columns() - pad - 1  # compensate for space after 'error:'
```

Comment:
This is kind of suspicious -- how will this work in the daemon? Should the client pass the terminal width to the server as part of each request, since the terminal may change between invocations?

Anyway, it might be better if this was provided by the caller.

Reply:
> This is kind of suspicious -- how will this work in the daemon?

I was going to fix this after the other PR lands. Essentially yes, this should be passed together with `is_tty` from the other PR.